app/validators/orphaned_node_validator.rb

```diff
@@ -0,0 +1,7 @@
+class OrphanedNodeValidator < ActiveModel::Validator
+  def validate(node)
+    unless node.parent.present?
+      node.errors.add :parent, 'Nodes must have parents'
+    end
+  end
+end
```

Comment:
Could we just validate `presence: true` on `:parent` instead of having a custom validator?